enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -164,7 +171,22 @@ const (
 	PublicKeyRootOfTrust         PolicyType = "PublicKey"
 	FulcioCAWithRekorRootOfTrust PolicyType = "FulcioCAWithRekor"
 	PKIRootOfTrust               PolicyType = "PKI"
-)
+
+    // Kuberentes secret data keys for `PublicKey` struct fields
+    SecretPublicKeyDataKey   = "public-key"
+    SecretRekorKeyDataKey    = "rekor-key"
+
+    // Kuberentes secret data keys for `PKI` struct fields (BYOPKI)
+    SecretPKIRootsDataKey           = "ca-roots"
+    SecretPKIIntermediatesDataKey   = "ca-intermediates"
+    SecretPKICertificateEmailKey    = "certificate-email"
+    SecretPKICertificateHostnameKey = "certificate-hostname"
+
+    // Kuberentes secret data keys for `FulcioCAWithRekor` struct fields
+    SecretFulcioCADataKey   = "fulcio-ca"
+    SecretRekorKeyDataKey   = "rekor-key" // reuse of existing constant
+    SecretFulcioOidcIssuerKey  = "oidc-issuer"
+    SecretFulcioSignedEmailKey = "signed-email"
```

Comment:
(A) … here we are storing non-cryptographic strings in the KMS. Is that (possible at all) and necessary?

E.g. I could well imagine that there would be quite a few policies within a company, all referencing the same Rekor keys and Fulcio/PKI CAs, but with a different certificate subject for each registry namespace (e.g. one certificate subject per a project team, signing builds of that project). It makes sense to manage the cryptographic keys in a specialized software, but the registry repo / certificate subject mapping is something a bit different.

This rather depends on the “Can the user still specify some values directly” question — if the answer to that is “yes”, and this is only an _option_ to store these parts of the policy in a Secret, I don’t see a reason to object. If the data had to be _only_ stored in a Secret, that intuitively feels unnatural to me.

Reply:
@mtrmac but would it be simpler for us to expect all the required data either coming from the k8s secret or entered manually by the user? I am concerned that allowing the image policy to get partially populated by k8s secret and user might lead to race conditions because it is unlikely both of those entities will update the policy at the same time. So maybe to keep things simpler, to begin with we mandate user to specify all the required fields of the specific policy in the k8s secret. 

But I agree that storing non-cryptographic information in KMS is little unnatural (but technically possible since it is just a key value pairs that you actually store there) but it does make it simple for us to manage that object,

Reply:
I don't have a strong opinion on this, as I'm not entirely sure about the use case for moving fields between the policy and the secret. That does seem rare to me, and I agree that race conditions might also be uncommon. That said, I lean towards having a single source of data for simplicity and to minimize potential issues.